tests/integration/test_charm_ipa.py

```diff
@@ -77,21 +104,64 @@ async def test_ipa_charm_has_ingress(ops_test: OpsTest):
 
     requirer_app_data = data.requirer.application_data
     model = dequote(requirer_app_data["model"])
+
     istio_ingress_address = await get_k8s_service_address(ops_test, "istio-ingress-k8s-istio")
 
     assert url == f"http://{istio_ingress_address}/{model}-ipa-tester"
 
-    url_parts = urlparse(url)
-    ip = url_parts.hostname
-    port = url_parts.port or 80
-    assert_can_connect(ip, port)
+
+@pytest.mark.abort_on_fail
+@pytest.mark.parametrize(
+    "external_hostname, expected_hostname",
+    [
+        ("foo.bar", "foo.bar"),  # Initial valid hostname
+        ("", None),  # Remove hostname
+        ("bar.foo", "bar.foo"),  # Change to a new valid hostname
+    ],
+)
+async def test_route_validity(
+    ops_test: OpsTest, external_hostname: str, expected_hostname: Optional[str]
+):
+    await ops_test.model.applications[APP_NAME].set_config(
+        {"external_hostname": external_hostname}
+    )
+    await ops_test.model.wait_for_idle([APP_NAME, "ipa-tester"])
+
+    data = get_relation_data(
+        requirer_endpoint="ipa-tester/0:ingress",
+        provider_endpoint="istio-ingress-k8s/0:ingress",
+        model=ops_test.model_full_name,
+    )
+
+    requirer_app_data = data.requirer.application_data
+    model = dequote(requirer_app_data["model"])
```

Comment:
I think this simplification works as well:
```suggestion
    model = ops_test.model.name
```

I've pushed a commit with this change, but feel free to change it back if I missed something